### PR TITLE
Spinning resin or laying eggs on vents prompts the user if they're sure

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
tweak: Aliens (and humans with alien organs) will have a prompt
if they try to create resin structures or lay eggs atop vents or scrubbers.
/:cl:

@WJohnston approved.

This stops stupid aliens from obscuring entrances to the very helpful
and useful atmospheric piping system. Unless they're really stupid and ignore the pop up message.

### CODE REVIEW

File: code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm [hunk 1]
```diff
@@ -254,18 +254,24 @@ Doesn't work on other aliens/AI.*/
 /obj/effect/proc_holder/alien/resin/fire(mob/living/carbon/user)
 	if(locate(/obj/structure/alien/resin) in user.loc)
 		to_chat(user, "<span class='danger'>There is already a resin structure there.</span>")
-		return 0
+		return FALSE
+
+	var/atom/movable/atmos_thing = locate(/obj/machinery/atmospherics/components/unary) in user.loc
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm]: could be simplified to `var/obj/machinery/atmospherics/components/unary/atmos_thing = locate() in user.loc`

File: code/modules/mob/living/carbon/alien/humanoid/queen.dm [hunk 2]
```diff
@@ -79,16 +79,22 @@
...
+	check_turf = TRUE
 	action_icon_state = "alien_egg"
 
 /obj/effect/proc_holder/alien/lay_egg/fire(mob/living/carbon/user)
 	if(locate(/obj/structure/alien/egg) in get_turf(user))
 		to_chat(user, "There's already an egg here.")
-		return 0
+		return FALSE
+
+	var/atom/movable/atmos_thing = locate(/obj/machinery/atmospherics/components/unary) in user.loc
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/alien/humanoid/queen.dm]: ditto

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm (hunk 1 above)
Comment (coiax) [reply code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm]: that seems to be saying the same thing, just differently.

Comment (ShizCalev) [reply code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm]: Just a touch cleaner imo

**Comment by ChangelingRain** [comment]
I don't like this, since it sometimes means you won't be able to block off corridors due to an object that isn't even solid.
It's basically a hamfisted solution to aliens being idiots.

**Review by ChangelingRain — CHANGES_REQUESTED** [review]
At least make it an "are you REALLY SURE about this" prompt instead of totally preventing it.
Otherwise it's just a clunky alien nerf that will randomly keep stupid aliens from doing something stupid.

**Comment by WJohn** [comment]
It's really not very hard to build around vents and scrubbers even as it is now. It doesn't affect smart players who already avoid them and incorporate them into their designs. It just prevents dumb ones from doing something dumb.

The rest is a mapping issue. Don't put vents and scrubbers immediately in front of the airlock, it's usually pretty bad at actually refilling or scrubbing a room.

**Comment by ChangelingRain** [comment]
Yes, smart players will consider vent placement, but sometimes you need to spin resin on vents and it's better to warn the player rather than end up fucking them over with a hardcoded restriction.

File: code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm [hunk 3]
```diff
@@ -254,18 +254,24 @@ Doesn't work on other aliens/AI.*/
...
+		return FALSE
 	if (!cost_check(check_turf,user))
-		return 0
+		return FALSE
 	to_chat(user, "<span class='notice'>You shape a [choice].</span>")
 	user.visible_message("<span class='notice'>[user] vomits up a thick purple substance and begins to shape it.</span>")
 
 	choice = structures[choice]
 	new choice(user.loc)
-	return 1
```
Comment (GunHog) [inline code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm]: This was a `1`, not a `0`. You set it to `FALSE`!

**Comment by Shadowlight213** [comment]
Why not make the vents be layered over the resin?
That seems like the simplest solution

**Comment by Cyberboss** [comment]
I agree with @ChangelingRain. @Shadowlight213 that'd look weird though

**Comment by coiax** [comment]
Okay, I'll change it to a "R U SURE???"

**Comment by coiax** [comment]
R U SUUURe

File: code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm [hunk 4]
```diff
@@ -59,6 +59,14 @@ Doesn't work on other aliens/AI.*/
 		return 0
 	return 1
 
+/obj/effect/proc_holder/alien/proc/check_vent_block(mob/living/user)
+	var/obj/machinery/atmospherics/components/unary/atmos_thing = locate() in user.loc
+	if(atmos_thing)
+		var/rusure = alert(user, "Laying eggs and shaping resin here would block access to [atmos_thing]. Do you want to continue?", "Blocking Atmospheric Component", "Yes", "No")
+		if(rusure == "No")
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm]: use != "Yes", alert may return null in exceptional circumstances

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Shadowlight213** [comment]
@Cyberboss Oh wait. I read this as weeds, not structures. nvm

**Comment by GunHog** [comment]
This is cool, now!

**Pull request merged**